lib/utils/include/utils/graph/undirected.h

```diff
@@ -9,7 +9,7 @@
 
 namespace FlexFlow {
 
-struct UndirectedEdge : public use_visitable_cmp<UndirectedEdge> {
```

Comment:
I generally prefer the explicit `public`--if you have an reason to prefer the other though I'd be happy to be convinced the other way